src/provider/provider_coarse.c

```diff
@@ -931,33 +925,36 @@ static umf_result_t coarse_memory_provider_initialize(void *params,
         coarse_provider->disable_upstream_provider_free = false;
     }
 
-    umf_result = coarse_memory_provider_set_name(coarse_provider);
+    umf_result_t umf_result = coarse_memory_provider_set_name(coarse_provider);
     if (umf_result != UMF_RESULT_SUCCESS) {
         LOG_ERR("name initialization failed");
         goto err_free_coarse_provider;
     }
 
+    // most of the error handling paths below set this error
+    umf_result = UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+
     coarse_provider->upstream_blocks =
         ravl_new_sized(coarse_ravl_comp, sizeof(ravl_data_t));
     if (coarse_provider->upstream_blocks == NULL) {
         LOG_ERR("out of the host memory");
-        umf_result = UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+        // umf_result is already set to UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY
```

Comment:
heh, so thanks to this little trick we'll never test this `if`, right...?

Reply:
No, I just removed one line from not covered code.

Reply:
I see, I was afraid that LOG is not consider a line to be covered (by the tool), but it is, so ok.